### DISCUSSION TITLE
Add shift-click support on all resource rows checkboxes

### DISCUSSION THREAD

**Discussion by jeanfrancoisgb** [laravel/nova-issues#4542] [Ideas & Feature Requests]
Hi,

Like Gmail and many online tools, it would be great that when we hold Shift, all checkboxes between the first and last one clicked are selected.

I did not find any other discussion about it.

To me it would be a huge quality of life improvement.

Example: If I click on the first checkbox, then hold Shift, and click on another checkbox in the bottom, all those between would then become checked.
![Screenshot - 2022-07-06T165546 203](https://user-images.githubusercontent.com/4010302/177641904-8ca64bdc-f942-434c-b58e-2ffd8d2f5840.png)

What do you think? I don't think adding this behavior would affect any user negatively since it would not cause regressions and let's be honest, we became used to it and are kind of expecting this behavior on modern tools.

**Comment by carestad** [comment]
I was just about to ask for the same thing. Highly wanted feature on my part since I quite oftenly want to select a range of resources within a index, but only a subset, and they are almost always right after each other. Shouldn't be too hard to implement either, nor would it, as @jeanfrancoisgb says, likely have a negative affect for existing users
